### PR TITLE
Add support for grpc based write proxy

### PR DESCRIPTION
**Pull request by gouthamve**
Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

### CODE REVIEW

**Comment by Dieterbe** [comment]
what's the status of this PR?

**Comment by Dieterbe** [comment]
@gouthamve what's the status of this.

**Comment by tomwilkie** [comment]
@Dieterbe we are running this in prod.  It need reviewing.

**Comment by tomwilkie** [comment]
I've given it a once over and LGTM

File: Gopkg.toml [hunk 1]
```diff
@@ -91,6 +96,15 @@ ignored = [
...
   version = "^2"
 
+#[[override]]
+  #name = "github.com/sercand/kuberesolver"
+  #version = "2.1.0"
+
+[[override]]
+  name = "github.com/sercand/kuberesolver"
+  source = "github.com/gouthamve/kuberesolver"
+  branch = "add-prom-metrics"
```
Comment (tomwilkie) [inline Gopkg.toml]: https://github.com/sercand/kuberesolver/pull/7 has been merged now, can we go back to upstream?

**Comment by tomwilkie** [comment]
One nit then get this merged and deployed please.

**Comment by Dieterbe** [comment]
> Gopkg.lock doesn't exist anymore for some reason

where? it exists in both this project and metrictank in current master.

**Comment by Dieterbe** [comment]
can you just rebase this PR on top of master?
I don't understand why this pr should include updates to sensitive vendored libraries.

File: Gopkg.toml (hunk 1 above)
Comment (gouthamve) [reply Gopkg.toml]: Done

**Comment by gouthamve** [comment]
@Dieterbe I've removed the update to `metrictank`, not sure what failed before. It was failing with:
```
➜  tsdb-gw git:(add-grpc-proxy) ✗ dep ensure        
grouped write of manifest, lock and vendor: failed to export github.com/grafana/metrictank: fatal: failed to unpack tree object 8a72e44d3c606112fb58f34b00b5ddc4377fe3c2
: exit status 128
```

But I've edited the revision for `metrictank` manually and it's works now.

**Comment by Dieterbe** [comment]
this PR makes changes to sarama and gopkg.in/ini.v1, why? sarama is sensitive and unrelated to your stuff.
also in general, please prefer rebasing on top of master, rather than merging master in your branch.
makes it much easier to review

**Comment by gouthamve** [comment]
@Dieterbe Pinned sarama to `1.19.0` as discussed.

**Comment by Dieterbe** [comment]
LGTM.
@robert-milan any objections to updating sarama from "v1.16.0" to "v1.19.0" (same as MT) ?

**Comment by robert-milan** [comment]
> LGTM.
> @robert-milan any objections to updating sarama from "v1.16.0" to "v1.19.0" (same as MT) ?

Sounds good as long as we aren't bumping the variables that control which kafka message format is actually used (`kafka-version`)

**Pull request merged**